packages/keycloak-user-management/src/components/CreateEditUser/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { Col, notification, Row } from 'antd';
+import { RouteComponentProps } from 'react-router';
+import * as Yup from 'yup';
+import { Store } from 'redux';
+import { connect } from 'react-redux';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import { HeaderBreadCrumb } from '../HeaderBreadCrumb';
+import {
+  fetchKeycloakUsers,
+  KeycloakUser,
+  makeKeycloakUsersSelector,
+  reducer as keycloakUsersReducer,
+  reducerName as keycloakUsersReducerName,
+  getAccessToken,
+} from '@opensrp/store';
+import { KeycloakService } from '@opensrp/keycloak-service';
+import Ripple from '../Loading';
+import { UserForm, UserFormProps } from '../../forms';
+import '../../index.css';
+
+reducerRegistry.register(keycloakUsersReducerName, keycloakUsersReducer);
+
+/** inteface for route params */
+
+export interface RouteParams {
+  userId: string;
+}
+
+/** props for editing a user view */
+export interface Props {
+  accessToken: string;
+  fetchKeycloakUsersCreator: typeof fetchKeycloakUsers;
+  keycloakUser: KeycloakUser | null;
+  serviceClass: typeof KeycloakService;
+}
+
+/** type intersection for all types that pertain to the props */
+export type PropsTypes = Props & RouteComponentProps<RouteParams>;
+
+/** default form initial values */
+
+export const defaultInitialValues: KeycloakUser = {
+  access: {
+    manageGroupMembership: false,
+    view: false,
+    mapRoles: false,
+    impersonate: false,
+    manage: false,
+  },
+  createdTimestamp: undefined,
+  disableableCredentialTypes: [],
+  email: '',
+  emailVerified: false,
+  enabled: true,
+  firstName: '',
+  id: '',
+  lastName: '',
+  notBefore: 0,
+  requiredActions: [],
+  totp: false,
+  username: '',
+};
+
+/** default props for editing user component */
+export const defaultProps: Partial<PropsTypes> = {
+  accessToken: 'hunter 2',
+  fetchKeycloakUsersCreator: fetchKeycloakUsers,
+  keycloakUser: null,
+  serviceClass: KeycloakService,
+};
+
+/** yup validations for practitioner data object from form */
+export const userSchema = Yup.object().shape({
+  lastName: Yup.string().required('Required'),
+  firstName: Yup.string().required('Required'),
+});
+
+const CreateEditUsers: React.FC<PropsTypes> = (props: PropsTypes) => {
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+  const { serviceClass, fetchKeycloakUsersCreator, keycloakUser, accessToken } = props;
+  const userId = props.match.params.userId;
+  const isEditMode = !!userId;
+  const initialValues = isEditMode ? keycloakUser : defaultInitialValues;
+  React.useEffect(() => {
+    if (userId) {
+      const serve = new serviceClass(
+        accessToken,
+        '/users',
+        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage'
```

Comment:
- Why are we hard coding these in the component?  They should be props
- Can we please remove magic strings?  We might need a constants module for this package

Reply:
created this issue to track this https://github.com/OpenSRP/web/issues/62